spec/services/coupon_service_spec.rb

```diff
@@ -3,67 +3,140 @@
-        expect(user.sticker_coupon).to be_a(StickerCoupon)
+      context 'there are both shirt and sticker coupons available' do
+        before do
+          FactoryBot.create(:shirt_coupon)
+          FactoryBot.create(:sticker_coupon)
+          coupon_service.assign_coupon
+        end
+
+        it 'assigns the user a shirt coupon' do
+          expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        end
+
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
 
-      it 'does not assign a shirt coupon' do
-        expect(user.shirt_coupon).to eq(nil)
+      context 'there are neither shirt nor sticker coupons available' do
+        before do
+          coupon_service.assign_coupon
+        end
+
+        it 'does not assign a shirt coupon' do
+          expect(user.shirt_coupon).to eq(nil)
+        end
+
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
     end
 
-    context 'there are both shirt and sticker coupons available' do
-      before do
-        FactoryBot.create(:shirt_coupon)
-        FactoryBot.create(:sticker_coupon)
-        coupon_service.assign_coupon
-      end
+    context 'with an incompleted user' do
+      let(:user) { FactoryBot.create(:user, :incompleted) }
+      let(:coupon_service) { CouponService.new(user) }
 
-      it 'assigns the user a shirt coupon' do
-        expect(user.shirt_coupon).to be_a(ShirtCoupon)
-      end
+      context 'there are only shirt coupons available' do
+        before do
+          FactoryBot.create(:shirt_coupon)
+          coupon_service.assign_coupon
+        end
+
+        # TODO: Add this test to allow gifting of extra shirt
+        # coupons in future Hacktoberfest events.
+        xit 'assigns the user a shirt coupon' do
+          expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        end
 
-      it 'does not assign a sticker coupon' do
-        expect(user.sticker_coupon).to eq(nil)
+        it 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
       end
-    end
 
-    context 'there are neither shirt nor sticker coupons available' do
-      before do
-        coupon_service.assign_coupon
+      context 'there are both shirt and sticker coupons available' do
+        before do
+          FactoryBot.create(:shirt_coupon)
+          FactoryBot.create(:sticker_coupon)
+          coupon_service.assign_coupon
+        end
+
+        # TODO: Add this test to allow gifting of extra shirt
+        # coupons in future Hacktoberfest events.
+        xit 'assigns the user a shirt coupon' do
+          expect(user.shirt_coupon).to be_a(ShirtCoupon)
+        end
+
+        # TODO: Add this test to allow gifting of extra shirt
+        # coupons in future Hacktoberfest events.
+        xit 'does not assign a sticker coupon' do
+          expect(user.sticker_coupon).to eq(nil)
+        end
```

Comment:
Pending tests to incorporate once the `CouponService.assign_coupon` functionality is adapted to be able to assign shirts to incomplete users as well